packages/webpack/src/plugin.js

```diff
@@ -221,11 +225,38 @@ class LavaMoatPlugin {
           JAVASCRIPT_MODULE_TYPE_ESM,
         ]
 
+        // Webpack has a concept of ignored modules
+        // When a module is ignored a carveout is necessary in policy enforcement for it because the ID that webpack creates for it is not exactly helpful.
+        // example outcome in the bundle: `const nodeCrypto = __webpack_require__(/*! crypto */ "?0b7d");`
+        // Sadly, even treeshaking doesn't eliminate that module. It's left there and failing to work when reached by runtime policy enforcement.
+        // Below is the most reliable way I've found to date to identify ignored modules.
+        /**
+         * @param {import('webpack').Module} m
+         * @returns {boolean}
+         */
+        const isIgnoredModule = (m) => {
+          if (
+            m.type === JAVASCRIPT_MODULE_TYPE_DYNAMIC &&
+            // @ts-expect-error BAD TYPES
+            m.identifierStr?.startsWith('ignored')
+          ) {
+            return true
+          }
+          // This function must be very narrow, because it's gatekeeping policy generation too
+          return false
+        }
```

Comment:
- How does the type of an `import('webpack').Module` differ from an "ignored module"?
- Are there any functions accepting either type but whose output differs depending on the type?

Reply:
This stuff would be much nicer if there was one more or one less layer of inheritance there. Now there are 2 or 3 reasonable categories of module types and I would like to select one but I can't do that based on types as they all inherit straight from Module which barely has anything on it.

I'm pretty sure this code will grow over time as we get more experience with real configurations of real projects.

Reply:
also, while I trust `process._rawDebug` won't magically disappear, I'm less trusting of webpack's ostensibly-undocumented  API surface (`identifierStr`?).  is there anything else we could use here?  can we maintain our own `WeakMap` instead?

Reply:
identifierStr is the least obscure way I found to check that a module is ignored. (ignored by webpack itself - specific internal webpack meaning)
This solution is relying on one much more internal thing and if webpack starts breaking things, we'll know (they're not keen on doing that tho)
I think we're good until webpack 6 but I'm planning to look into a less tight integration with webpack over time, opening up the possibility we could support rspack too.